src/containers/Yields/utils.ts

```diff
@@ -1,23 +1,47 @@
 import { calculateLoopAPY, YieldsData } from '~/containers/Yields/queries/index'
 import { attributeOptions, attributeOptionsMap } from './Filters/Attributes'
 
+interface IToFilterPool {
+	curr: YieldsData['props']['pools'][number]
+	selectedProjectsSet: Set<string>
+	selectedChainsSet: Set<string>
+	selectedAttributes: string[]
+	includeTokens: string[]
+	exactTokens: string[]
+	selectedCategoriesSet: Set<string>
+	excludeTokensSet: Set<string>
+	pathname: string
+	minTvl: number | null
+	maxTvl: number | null
+	minApy: number | null
+	maxApy: number | null
+	pairTokens: string[]
+	usdPeggedSymbols: string[]
+}
+
 export function toFilterPool({
 	curr,
-	selectedProjects,
-	selectedChains,
+	selectedProjectsSet,
+	selectedChainsSet,
 	selectedAttributes,
 	includeTokens,
-	excludeTokens,
+	excludeTokensSet,
 	exactTokens,
-	selectedCategories,
+	selectedCategoriesSet,
 	pathname,
 	minTvl,
 	maxTvl,
 	minApy,
 	maxApy,
 	pairTokens,
 	usdPeggedSymbols
-}) {
+}: IToFilterPool) {
+	const tokensInPoolArray = curr.symbol
```

Comment:
Great PR, makes code a lot cleaner!

My only concern is this expression, it's a bit hard to read, and `.split('(')[0]` might not work as expected in some corner cases I guess? Found this [weird pool](https://defillama.com/yields/pool/33ed47c5-37ea-43b2-a5d1-6a46b1fcc73f) as example, is it ok if we throw away everyting right to `(`?
<img width="555" height="84" alt="Screenshot 2025-11-28 at 19 16 10" src="https://github.com/user-attachments/assets/d7105e0b-c942-47e3-ba73-a4417c90f4b7" />

Reply:
(correct me if I'm wrong, maybe this is completely irrelevant)